src/Network/Transport/TCP.hs

```diff
@@ -870,7 +877,9 @@ handleConnectionRequest transport sock = handle handleException $ do
     forM_ (tcpUserTimeout $ transportParams transport) $
       N.setSocketOption sock N.UserTimeout
     ourEndPointId <- recvInt32 sock
-    theirAddress  <- EndPointAddress . BS.concat <$> recvWithLength sock
+    let maxAddressLength = tcpMaxAddressLength $ transportParams transport
```

Comment:
I think actually we could still have a max length here, even when the user does not care. The endpoint address size is bounded by the max of {IPv4, IPv6, DNS name} + port number + local endpoint id + separators. This is something like `maximum [15, 45, 250] + 5 + 10 + 3` which is 268.